module1-introduction-to-backend/r2.1-diving-into-rest-apis/README.md

```diff
@@ -0,0 +1,122 @@
+# Diving into REST APIs
+HTTP messages are how data is exchanged between a server and a client. There are two types of messages: requests sent by the client to trigger an action on the server, and responses, the answer from the server. The learning objectives of this lesson are:
+1. Understanding HTTP request types
+2. Understanding HTTP request and response structure
+3. Understanding HTTP response status codes
+
+## HTTP Requests
+In the previous lesson, we briefly talked about GET, POST, PUT and DELETE request methods.
+
+HTTP defines a set of request methods to indicate the desired action to be performed for a given resource. This set includes:
+- GET: Requests a representation of the specified resource. Requests using GET should only retrieve data.
```

Comment:
@osamaakb i think you need to trim your section if some of this is already explained